bundles/org.openhab.binding.tplinkrouter/src/main/java/org/openhab/binding/tplinkrouter/internal/TpLinkRouterTelnetConnector.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.tplinkrouter.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.InterruptedIOException;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TpLinkRouterTelnetConnector} is responsible of the telnet connection.
+ *
+ * @author Olivier Marceau - Initial contribution
+ */
+@NonNullByDefault
+public class TpLinkRouterTelnetConnector {
+
+    private static final Integer TIMEOUT = 60000; // 1 minute
+
+    private final Logger logger = LoggerFactory.getLogger(TpLinkRouterTelnetConnector.class);
+
+    private @Nullable Thread telnetClientThread;
+    private @Nullable Socket socket; // use raw socket since commons net usage seems discouraged
+    private @Nullable OutputStreamWriter out;
+
+    public void connect(TpLinkRouterTelenetListener listener, TpLinkRouterConfiguration config, String thingUID)
+            throws IOException {
+        logger.debug("Connecting to {}", config.hostname);
+
+        Socket socketLocal = new Socket();
+        socketLocal.connect(new InetSocketAddress(config.hostname, config.port));
+        socketLocal.setSoTimeout(TIMEOUT);
+        socketLocal.setKeepAlive(true);
+
+        InputStreamReader inputStream = new InputStreamReader(socketLocal.getInputStream());
+        this.out = new OutputStreamWriter(socketLocal.getOutputStream());
+        this.socket = socketLocal;
+        loginAttempt(listener, inputStream, config);
+        Thread clientThread = new Thread(() -> listenInputStream(listener, inputStream, config));
+        clientThread.setName("OH-binding-" + thingUID);
+        this.telnetClientThread = clientThread;
+        clientThread.start();
+        logger.debug("TP-Link router telnet client connected to {}", config.hostname);
+    }
+
+    public void dispose() {
+        logger.debug("disposing connector");
+        Thread clientThread = telnetClientThread;
+        if (clientThread != null) {
+            clientThread.interrupt();
+        }
+        Socket socketLocal = socket;
+        if (socketLocal != null) {
+            try {
+                socketLocal.close();
+            } catch (IOException e) {
+                logger.error("Error while disconnecting telnet client", e);
```

Comment:
DEBUG level is sufficient here.
Or WARN level if you think it should be seen by any user.